sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java

```diff
@@ -140,6 +141,11 @@ public final class OpenAIAsyncClient {
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<BinaryData>> getEmbeddingsWithResponse(String deploymentOrModelName,
         BinaryData embeddingsOptions, RequestOptions requestOptions) {
+        try {
+            embeddingsOptions = addEncodingFormat(embeddingsOptions);
+        } catch (JsonProcessingException e) {
+            return Mono.error(e);
```

Comment:
If we are throwing RuntimeError in sync client, we should do the same in async client too.